lib/internal/promisify.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+module.exports = promisify;
+
+function promisify(fn, single) {
+  Object.defineProperty(inner, 'name', {
+    value: fn.name + 'Promisified'
+  });
+  return inner;
+
+  function inner() {
+    if (typeof arguments[arguments.length - 1] === 'function') {
+      return fn.apply(this, arguments);
```

Comment:
Why? `.apply()` used in conjuction with `arguments` transforms to very efficient code in V8 as it doesn't need to recreate argument array. If you were thinking about deopt, it's different in your case because you pass `arguments` to nested function, while `inner` uses own `arguments`.